cli/internal/cloudcmd/upgrade.go

```diff
@@ -115,13 +115,24 @@ func (u *Upgrader) UpgradeNodeVersion(ctx context.Context, conf *config.Config)
 	upgradeErrs := []error{}
 	upgradeErr := &compatibility.InvalidUpgradeError{}
 	err = u.updateImage(&nodeVersion, imageReference, imageVersion.Version)
-	if errors.As(err, &upgradeErr) {
+	switch {
+	case errors.As(err, &upgradeErr):
 		upgradeErrs = append(upgradeErrs, fmt.Errorf("skipping image upgrades: %w", err))
+	case err != nil:
+		return fmt.Errorf("updating image version: %w", err)
 	}
 
 	components, err := u.updateK8s(&nodeVersion, versionConfig.ClusterVersion, versionConfig.KubernetesComponents)
-	if errors.As(err, &upgradeErr) {
+	switch {
+	case errors.As(err, &upgradeErr):
 		upgradeErrs = append(upgradeErrs, fmt.Errorf("skipping Kubernetes upgrades: %w", err))
+	case err != nil:
+		return fmt.Errorf("updating Kubernetes version: %w", err)
+	case err == nil:
+		err := u.applyComponentsCM(ctx, components)
+		if err != nil {
+			return fmt.Errorf("applying k8s components ConfigMap: %w", err)
+		}
```

Comment:
What do you think about:
```suggestion
	case err == nil:
		err := u.applyComponentsCM(ctx, components)
		if err != nil {
			return fmt.Errorf("applying k8s components ConfigMap: %w", err)
		}
	case errors.As(err, &upgradeErr):
		upgradeErrs = append(upgradeErrs, fmt.Errorf("skipping Kubernetes upgrades: %w", err))
	default:
		return fmt.Errorf("updating Kubernetes version: %w", err)
```

I think having the `err==nil` case is a bit worse, but I prefer having a `default` which might makes it clearer that this is the "catch everything unexpected" case which should never be deleted. But I have no strong opinion about this.